### ISSUE TITLE
Unable to extract uploader id.

### ISSUE THREAD

**Issue by holonen** [ytdl-org/youtube-dl#32449]
## Checklist

- [x] I'm reporting a broken site support issue
- [x] I've verified that I'm running youtube-dl version **2021.12.17**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar bug reports including closed ones
- [ ] I've read bugs section in FAQ

## Verbose log

```
(base) amitsen@MacBook-Pro-som-tillhor-Amit 2023 % youtube-dl --verbose -x https://youtu.be/P-2P3MSZrBM

[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--verbose', '-x', 'https://youtu.be/P-2P3MSZrBM']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Git HEAD: 9d01e1dea
[debug] Python version 3.10.10 (CPython) - macOS-13.3.1-arm64-arm-64bit
[debug] exe versions: none
[debug] Proxy map: {}
[youtube] P-2P3MSZrBM: Downloading webpage
ERROR: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/youtube_dl/YoutubeDL.py", line 815, in wrapper
    return func(self, *args, **kwargs)
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/youtube_dl/YoutubeDL.py", line 836, in __extract_info
    ie_result = ie.extract(url)
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/youtube_dl/extractor/common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/youtube_dl/extractor/youtube.py", line 1794, in _real_extract
    'uploader_id': self._search_regex(r'/(?:channel|user)/([^/?&#]+)', owner_profile_url, 'uploader id') if owner_profile_url else None,
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/youtube_dl/extractor/common.py", line 1012, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

(base) amitsen@MacBook-Pro-som-tillhor-Amit 2023 % 
```

## Description

WRITE DESCRIPTION HERE
Issue Description:

I encountered an error while trying to download a video using YouTube-DL. The error message states "Unable to extract uploader id." This problem prevents me from successfully downloading the video.

Additional Information:

YouTube-DL version: 2021.12.17
Operating System: macOS-13.3.1-arm64-arm-64bit
Python version: 3.10.10 (CPython)
Command Used: youtube-dl --verbose -x https://youtu.be/P-2P3MSZrBM
Steps to Reproduce:

Run the command: youtube-dl --verbose -x https://youtu.be/P-2P3MSZrBM
Observe the error message "Unable to extract uploader id."
Expected Behavior:

I expected the video to be downloaded successfully without any errors.

Additional Context:

Video URL: https://youtu.be/P-2P3MSZrBM
YouTube Channel or User: N/A
Verbose Output:

csharp
Copy code
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--verbose', '-x', 'https://youtu.be/P-2P3MSZrBM']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Git HEAD: 9d01e1dea
[debug] Python version 3.10.10 (CPython) - macOS-13.3.1-arm64-arm-64bit
[debug] exe versions: none
[debug] Proxy map: {}
[youtube] P-2P3MSZrBM: Downloading webpage
ERROR: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug. Make sure you are using the latest version; see https://yt-dl.org/update on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/youtube_dl/YoutubeDL.py", line 815, in wrapper
    return func(self, *args, **kwargs)
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/youtube_dl/YoutubeDL.py", line 836, in __extract_info
    ie_result = ie.extract(url)
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/youtube_dl/extractor/common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/youtube_dl/extractor/youtube.py", line 1794, in _real_extract
    'uploader_id': self._search_regex(r'/(?:channel|user)/([^/?&#]+)', owner_profile_url, 'uploader id') if owner_profile_url else None,
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/youtube_dl/extractor/common.py", line 1012, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract uploader id; please report this issue on https://yt-dl.org/bug. Make sure you are using the latest version; see https://yt-dl.org/update on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Suggested Solution:

Based on my research, this issue may be related to changes in YouTube's website structure. It's possible that an update to the YouTube-DL tool is needed to address this problem.

Account Credentials:

No account credentials are required for reproducing the issue.

**Comment by dirkf** [comment]
>[x] I've searched the bugtracker for similar bug reports including closed ones

Learn how by reading #30839 (**again**?).

>[ ] I've read bugs section in FAQ

Read and apply.

**Comment by holonen** [comment]
now read the bug section in FAQ

**Comment by holonen** [comment]
Solved by commenting out this line in youtube.py
        #    'uploader_id': self._search_regex(r'/(?:channel|user)/([^/?&#]+)', owner_profile_url, 'uploader id') if owner_profile_url else None,

**Comment by dirkf** [comment]
Just install the up-to-date code instead. Otherwise the other fixed issues listed in #30839 will still apply, not to mention all the ones since 2021.12. The master issue -- linked from #30839 (you did read through that ?) -- suggests `brew install --head ...`.
